Feature Media Controller

aliyahmoore opened the pull request:
### **Description**
- **What does this PR do?**
Enhanced the Media MVC and implemented easy navigation to the media page via the navbar and the "Latest News" section on the homepage.

---

### **Key Changes (if applicable, delete if not applicable)**
1. **Models:**
   - [ ] Created a new model Medium (medium.rb) (Rails changed it to medium)
   - [ ] Ran the migration to set up the database schema for the Medium model.
   - [ ] Configured validations for the model.

2. **Controllers:**
   - [ ] Updated MediaController with index, and CRUD actions.
   - [ ] Added set_medium to be used only for update and destroy actions.
   - [ ] Added medium_params to permit only name, description, and link.

3. **Views:**
   - [ ] Added a view that generates a list of news events, including descriptions and links, featuring the founder or any event related to Amanahrec.
   - [ ] Added a "Latest News" tab in the navbar and a attached a link to the "Latest News" section on the homepage. Both linking to the media page for easy navigation.
   - [ ] Added a form for creating a new media post. 
   - [ ] Added an edit button to each media post so once it's clicked it will toggle down an edit form for that specific media post. 

4. **Routes:**
   - [ ] Added resources :media to define standard RESTful routes for media resources.
   
5. **Gems:**
   - [ ] No gems added.

6. **Tests:**
   - [ ] Verified that the title and image are properly centered on the page.
   - [ ] Tested saving data to the database using Rails Console.
   - [ ] Tested navigation to the media page via the navbar tab and the "Latest News" section on the homepage.
   - [ ] Ran RuboCop and confirmed no offenses were detected.

---

### **How to Test**
1. **Setup:**
   - [ ] Run migrations: `rails db:migrate`
   - [ ] Test saving data to the database using Rails Console.
   - [ ] Test navigation to the media page from the navbar and "Latest News" section. 
---

### **Screenshots**

<img width="1433" alt="Media" src="https://github.com/user-attachments/assets/59a8a0d5-39be-4821-8c2a-f955047cb3c3" />

aliyahmoore commented:
Check latest commit: "Implemented CRUD forms for media view".  

<img width="1083" alt="Screenshot 2025-01-20 at 10 18 52 PM" src="https://github.com/user-attachments/assets/bd058840-e22a-4e18-ac91-a5475bcd64c5" />

nadiakane26 commented:
Looks good and functioning. I was confused by some what the PR said such as _Added a "Latest News" tab in the navbar and a attached a link to the "Latest News" section on the homepage. Both linking to the media page for easy navigation._  and _Test navigation to the media page from the navbar and "Latest News" section._

nadiakane26 commented on app/assets/stylesheets/media.css:
Can you use bootstrap instead?

nadiakane26 commented on app/javascript/application.js:
What does this do?

nadiakane26 commented on app/javascript/media_toggle.js:
Was there a toggle that I missed somewhere?

nadiakane26 commented on app/views/layouts/_navbar.html.erb:
Update the path from the routes. You can check the routes by running `rails routes | grep media`

nadiakane26 commented on app/views/media/index.html.erb:
If this gets too long, think of using a partial especially for the form.

nadiakane26 commented on config/importmap.rb:
Again, not sure where this pops up

nadiakane26 reviewed (changes requested):
The CRUD functions work, there's a few minor improvements needed. Good job!

aliyahmoore commented:
Made changes/updates to the view by incorporating bootstrap. 
<img width="1401" alt="Screenshot 2025-01-21 at 11 35 16 AM" src="https://github.com/user-attachments/assets/bd07c1f7-0b73-46dd-ac5d-fc65941d5fb7" />

aliyahmoore commented:
Added published_date and media_link columns to Media table, ran migrations, and updated MVC accordingly.
<img width="1410" alt="Screenshot 2025-01-21 at 2 31 25 PM" src="https://github.com/user-attachments/assets/997614b3-7881-4fa9-945d-7096a001f35d" />

nadiakane26 commented on app/views/media/index.html.erb:
```diff
@@ -0,0 +1,114 @@
+                            data: { medium_id: medium.id } %>
+              </div>
+            </div>
+            <p class="media-description mb-4"><%= medium.description %></p>
```
I don't think description is needed. If it's optional, it should not show if there is nothing in the description. 
![image](https://github.com/user-attachments/assets/7e2539d8-f3b5-4f55-9607-88d0cb6ce97d)

nadiakane26 commented on app/views/media/index.html.erb:
```diff
@@ -0,0 +1,114 @@
+              </div>
+            </div>
+            <p class="media-description mb-4"><%= medium.description %></p>
+            <p><strong>Organization:</strong> <%= medium.organization_name %></p>
```
Maybe instead of organization say news organization?

nadiakane26 commented on app/views/media/index.html.erb:
Can you order by date it's published consider using bootstrap and `will_paginate` gem

nadiakane26 replied:
Is there a way to embed a snippet of the article?

nadiakane26 commented on app/views/media/index.html.erb:
```diff
@@ -0,0 +1,114 @@
+    <h2 class="text-dark">Add New Media Post</h2>
+  </div>
+
+  <%= form_with model: @medium, url: media_path, local: true do |f| %>
```
This should be a partial

nadiakane26 commented on app/views/media/index.html.erb:
```diff
@@ -0,0 +1,114 @@
+                            data: { confirm: 'Are you sure you want to delete this media?' } %>
+            </div>
+            
+            <div class="edit-form-container mt-4" id="edit-form-<%= medium.id %>" style="display: none;">
```
Since you're using the same form to edit and create new one it should be a partial, the edit doesn't have the same fields as the new

rukia-nuviz commented on db/migrate/20250121202124_add_published_date_and_organization_name_to_media.rb:
```diff
@@ -0,0 +1,6 @@
+class AddPublishedDateAndOrganizationNameToMedia < ActiveRecord::Migration[7.2]
+  def change
+    add_column :media, :published_date, :date
+    add_column :media, :organization_name, :string
```
this should all be in the same migration since it is in the same PR. it should be in the `create_media.rb` migration. Ask @peterlee597 for help, I gave him the same feedback

rukia-nuviz commented on db/migrate/20250116193437_create_media.rb:
```diff
@@ -0,0 +1,11 @@
+class CreateMedia < ActiveRecord::Migration[7.2]
+  def change
+    create_table :media do |t|
+      t.string :name
```
are any of these fields required?

rukia-nuviz commented on app/models/medium.rb:
```diff
@@ -0,0 +1,3 @@
+class Medium < ApplicationRecord
+  validates :name, :link, :published_date, :organization_name, presence: true
```
why is this class and file called `medium.rb` it should be called `media.rb` and `class Media` ?

rukia-nuviz commented on app/views/media/_form.html.erb:
```diff
@@ -0,0 +1,25 @@
+<%= form_with model: medium, local: true do |f| %>
+  <div class="mb-3">
+    <%= f.label :name, class: "form-label text-dark" %>
+    <%= f.text_field :name, required: true, class: "form-control" %>
```
add `required: true` to all fields that are required so `*` shows in form

rukia-nuviz commented on app/controllers/media_controller.rb:
```diff
@@ -0,0 +1,49 @@
+
+  def index
+    @media = Medium.all
+    @medium = Medium.new
```
what is medium? and why are you calling Medium.new in index action?

rukia-nuviz commented on app/controllers/media_controller.rb:
```diff
@@ -0,0 +1,49 @@
+  end
+
+  def create
+    @medium = Medium.new(medium_params)
```
why is this called `medium` but controller is called `media`?

rukia-nuviz reviewed (changes requested):
left feedback
